includes/admin/class-amp-post-meta-box.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * AMP meta box settings.
+ *
+ * @package AMP
+ * @since 0.6
+ */
+
+/**
+ * Post meta box class.
+ *
+ * @since 0.6
+ */
+class AMP_Post_Meta_Box {
+
+	/**
+	 * Assets handle.
+	 *
+	 * @since 0.6
+	 * @const string
+	 */
+	const ASSETS_HANDLE = 'amp-post-meta-box';
+
+	/**
+	 * The post meta key.
+	 *
+	 * @since 0.6
+	 * @const string
+	 */
+	const POST_META_KEY = 'amp_status';
+
+	/**
+	 * The nonce name.
+	 *
+	 * @since 0.6
+	 * @const string
+	 */
+	const NONCE_NAME = 'amp-status';
+
+	/**
+	 * The nonce action.
+	 *
+	 * @since 0.6
+	 * @const string
+	 */
+	const NONCE_ACTION = 'amp-update-status';
+
+	/**
+	 * Initialize.
+	 *
+	 * @since 0.6
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_assets' ) );
+		add_action( 'post_submitbox_misc_actions', array( $this, 'render_status' ) );
+		add_action( 'save_post', array( $this, 'save_amp_status' ) );
+	}
+
+	/**
+	 * Enqueue admin assets.
+	 *
+	 * @since 0.6
+	 */
+	public function enqueue_admin_assets() {
+		$post = get_post();
+
+		// Stop if the post doesn't have AMP support.
+		if ( ! isset( $post->post_type ) || true !== post_type_supports( $post->post_type, AMP_QUERY_VAR ) ) {
+			return;
+		}
+
+		// Styles.
+		wp_enqueue_style(
+			self::ASSETS_HANDLE,
+			amp_get_asset_url( 'css/amp-post-meta-box.css' ),
+			false,
+			AMP__VERSION
+		);
+
+		// Scripts.
+		wp_enqueue_script(
+			self::ASSETS_HANDLE,
+			amp_get_asset_url( 'js/amp-post-meta-box.js' ),
+			array( 'jquery' ),
+			AMP__VERSION
+		);
+		wp_add_inline_script( self::ASSETS_HANDLE, sprintf( 'AmpPostMetaBox.boot( %s );',
+			wp_json_encode( array(
+				'previewLink' => esc_url_raw( add_query_arg( AMP_QUERY_VAR, true, get_preview_post_link( $post ) ) ),
+			) )
+		) );
+	}
+
+	/**
+	 * Render AMP status.
+	 *
+	 * @since 0.6
+	 * @param object $post \WP_POST object.
+	 */
+	public function render_status( $post ) {
+		$verify = (
+			isset( $post->ID )
+			&&
+			isset( $post->post_type )
+			&&
+			post_type_supports( $post->post_type, AMP_QUERY_VAR )
```

Comment:
This differ from the acceptance criteria, I would suggest to add the ability to disable AMP on a per post basis for all post types which have post type support set, not only pages (especially now that WordPress Core has [Post Type Templates)](https://make.wordpress.org/core/2016/11/03/post-type-templates-in-4-7/). Thoughts @amedina @westonruter?

Reply:
This differ from from the acceptance criteria (point 9) in a sense that the AMP status option would not be displayed at all if the post type does not have AMP support set to avoid visual noise. The thinking behind this is that the AMP status (enabled/disabled) should not be editable for post types (page included) if it doesn't have AMP support set.

If we really want to show the status as disabled, one approach would be to change to edit section with a text indicating that AMP post type support is not enabled for the current post with a link to the AMP settings page where user may enable the post type support.

Thoughts @amedina @westonruter?

Reply:
@ThierryA I think you're right. The AMP status toggle shouldn't be shown at all if the post type doesn't support AMP to begin with.